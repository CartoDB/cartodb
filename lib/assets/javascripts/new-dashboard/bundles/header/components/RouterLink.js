import Vue from 'vue';
import getCARTOData from 'new-dashboard/store/utils/getCARTOData';

const CARTOData = getCARTOData();

const RouterLink = Vue.component(
  'router-link',
  {
    template: '<a :href="url"><slot/></a>',
    props: {
      staticRoute: String
    },
    computed: {
      url: function () {
        const baseUrl = CARTOData.user_data.base_url;
        return baseUrl + this.staticRoute;
      }
    }
  }
);

export default RouterLink;
