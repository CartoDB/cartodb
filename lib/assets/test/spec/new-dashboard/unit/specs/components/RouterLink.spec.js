import { shallowMount } from '@vue/test-utils';
import RouterLink from 'new-dashboard/bundles/header/components/RouterLink';

jest.mock('new-dashboard/store/utils/getCARTOData', () => function () {
  return {
    user_data: {
      base_url: 'example-baseurl.com'
    }
  };
});

describe('RouterLink.js', () => {
  it('should a link element with the given static route as url', () => {
    const routerLinkElement = shallowMount(RouterLink, {
      propsData: {
        staticRoute: '/test-path'
      },
      slots: { default: 'Test Router Link' }
    });
    expect(routerLinkElement).toMatchSnapshot();
  });
});
