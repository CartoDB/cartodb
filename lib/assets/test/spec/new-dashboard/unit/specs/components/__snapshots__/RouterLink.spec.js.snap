// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterLink.js should a link element with the given static route as url 1`] = `<a href="example-baseurl.com/test-path">Test Router Link</a>`;
